### PR TITLE
setup and doc pre-commit

### LINKED ISSUE #5: precommit to clean notebooks?
should we configure precommit to clean all notebooks in ``gallery/interactive`` before commit?
in this way it would be clean

### PR DESCRIPTION
**Pull request by ahms5**
closes #5

``pre-commit install`` is required to execute, therefore I added a Get started section in the contribution.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -25,6 +25,53 @@ The rendered version of the gallery is hosted on `readthedocs.org`_.
...
+
+   Now you can make your changes locally.
+
+6. When you're done making changes, check that your changes pass flake8 and the
+   tests::
+
+    $ flake8 pyfar tests
+    $ pytest
+
+   flake8 test must pass without any warnings for `./pyfar` and `./tests` using the default or a stricter configuration. Flake8 ignores `E123/E133, E226` and `E241/E242` by default. If necessary adjust the your flake8 and linting configuration in your IDE accordingly.
```
Comment (f-brinkmann) [inline README.rst:62]: I think, we don't need this here - or only for tests, once we have some.

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks, looks fine, I only had one comment.

**Review by pingelit — APPROVED** [review]

File: README.rst [hunk 2]
```diff
@@ -25,6 +25,45 @@ The rendered version of the gallery is hosted on `readthedocs.org`_.
...
+
+    $ git clone https://github.com/YOUR_USERNAME/gallery.git
+    $ cd gallery
+
+3. Install your local copy into a virtualenv. Assuming you have Anaconda or Miniconda installed, this is how you set up your fork for local development::
+
+    $ conda create --name gallery python
+    $ conda activate gallery
+    $ pip install -e .
+    $ pip install -r requirements_dev.txt
```
Comment (pingelit) [inline README.rst:44]:
```suggestion
    $ pip install -r requirements.txt
```
I just noticed, the previous version does not work.

**Review by pingelit — CHANGES_REQUESTED** [review]
I just notices some errors.

**Comment by pingelit** [comment]
Thanks, I'm also getting a warning with the pre-commit:
```bash
[WARNING] The 'files' field in hook 'jupyter-nb-clear-output' is a regex, not a glob -- matching '/*' probably isn't what you want here
```
Are you getting this as well?

**Review by mberz — CHANGES_REQUESTED** [review]
I'm getting the same warning as @pingelit 

``` console
[WARNING] The 'files' field in hook 'jupyter-nb-clear-output' is a regex, not a glob -- matching '/*' probably isn't what you want here
jupyter-nb-clear-output..............................(no files to check)Skipped
```

It fails to clear notebooks with cell output for me since it's not detecting files properly.

Running nbconvert on the respective files manually works as expected.

**Comment by ahms5** [comment]
If you want to commit a notebook with outputs, it fails and delete the outputs, then you just need to add the changes and it works.


For more detail:
https://zhauniarovich.com/post/2020/2020-06-clearing-jupyter-output/

File: README.rst [hunk 3]
```diff
@@ -25,6 +25,44 @@ The rendered version of the gallery is hosted on `readthedocs.org`_.
...
+
+    $ pre-commit install
+
+5. Create a branch for local development. Indicate the intention of your branch in its respective name (i.e. `feature/branch-name` or `bugfix/branch-name`)::
+
+    $ git checkout -b name-of-your-bugfix-or-feature
+
+   Now you can make your changes locally.
+
+6. Commit your changes and push your branch to GitHub::
```
Comment (f-brinkmann) [inline README.rst:55]: Maybe add information about the pre-commit workflow:

'Note that the commit will fail if you are trying to commit notebooks including output. Trying to push will automatically remove the output in these notebooks after which they must be saved and the push can be done.'

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks, works for me as well. I would only suggest to add some information about the error that occurs, when trying to push a notebook including output.

**Review by f-brinkmann — APPROVED** [review]

**Pull request merged**